### PR TITLE
RM-19979 Fix that Ctrie iterator did not return entries from tNodes.

### LINKED ISSUE #171: Ctrie Iterator doesn't return entries from tNodes
The `traverse()` function only covers `main.cNode` and `main.lNode`.  If the `cNode` has been collapsed to a `tNode` then its entry is skipped.

I'll put up a PR to fix.

### PR DESCRIPTION
**Pull request by fasaxc**
Fixes #171

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Raven
<details>
<summary>Number of Findings: 0</summary>

</details>

**Comment by dzyp** [comment]
+1

**Comment by tylertreat** [comment]
lgtm

File: trie/ctrie/ctrie.go [hunk 1]
```diff
@@ -405,6 +405,8 @@ func (c *Ctrie) traverse(i *iNode, ch chan<- *Entry, cancel <-chan struct{}) err
 				return errCanceled
 			}
 		}
+	case main.tNode != nil:
+		ch <- main.tNode.Entry
```
Comment (trevorgray) [inline trie/ctrie/ctrie.go]: To support cancellation this should be:

```
select {
case ch <- main.tNode.Entry:
case <-cancel:
	return errCanceled
}
```

**Comment by fasaxc** [comment]
@trevorgray I've pushed the requested markups.

**Comment by stevenosborne-wf** [comment]
@fasaxc Looks good other than the change to the copyright. Please remove that change and we can get this merged.

**Comment by brianshannan-wf** [comment]
+1

**Comment by dustinhiatt-wf** [comment]
+1

**Comment by teresarevious-wf** [comment]
RM +1 verified dependencies.

**Pull request merged**